rotkehlchen/api/rest.py

```diff
@@ -3828,12 +3828,16 @@ def download_database_backup(self, filepath: Path) -> Response:
         )
 
     @require_loggedin_user()
-    def delete_database_backup(self, filepath: Path) -> Response:
-        if filepath.parent != self.rotkehlchen.data.db.user_data_dir:
-            error_msg = f'DB backup file {filepath} is not in the user directory'
-            return api_response(wrap_in_fail_result(error_msg), status_code=HTTPStatus.CONFLICT)
+    def delete_database_backup(self, files: List[Path]) -> Response:
+        for filepath in files:
+            if filepath.parent != self.rotkehlchen.data.db.user_data_dir:
```

Comment:
Hmmm this is for a really edge case and should not happen in real life BUT I think you should first check all given files for this and return an error as you do if this happens.

Then iterate over all files again, and unlink them.

This is to avoid the super edge case where 1 file is deleted and next file has this error and the endpoint call returns an error while actually having deleted 1 file.